### PR TITLE
feat(plugins): add keycloak plugin

### PR DESCRIPTION
**Pull request by Shurtu-gal**
Fixes: amplication/amplication#7044

## PR Details

Add Keycloak plugin for authentication

## PR Checklist

- [x] Make the plugin
- [x] Add README
- [x] Tests for the changes have been added
- [x] `npm build` doesn't throw any error
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by overbit** [comment]
@Shurtu-gal do you mind rebasing on master as we just merged a PR that clean all linting/testing issues?

**Comment by Shurtu-gal** [comment]
Done @overbit. What about templates, I guess they should also be put in `.eslintignore`

**Comment by overbit** [comment]
@Shurtu-gal that is a good point. Can you raise a new PR just for that (eslint ignore in the template)?

**Comment by Shurtu-gal** [comment]
Sure.

**Review by overbit — CHANGES_REQUESTED** [review]
@Shurtu-gal really good work. Just a couple of comments but generally looks great

File: plugins/auth-keycloak/src/templates/jwt.strategy.base.template.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { UnauthorizedException } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { PassportStrategy } from "@nestjs/passport";
+import { passportJwtSecret } from "jwks-rsa";
+import { ExtractJwt, Strategy } from "passport-jwt";
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply plugins/auth-keycloak/src/templates/jwt.strategy.base.template.ts:6]: They are being being removed from the generated code out [here](https://github.com/Shurtu-gal/plugins/blob/973c2794fedb448b29a82a87d4d321409697b30c/plugins/auth-keycloak/src/core/create-jwt-strategy-base.ts#L110)

File: plugins/auth-keycloak/src/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import { Entity, EntityField, EnumDataType } from "@amplication/code-gen-types";
+import { EmailError, placeholders } from "../constants";
+import { IRecipe } from "../types";
+import { Settings } from "../types";
+
+export function pascalCase(str: string): string {
+  return str
+    .replace(/^[a-z]/, (m) => m.toUpperCase())
+    .replace(/-([a-z])/g, (m) => m[1].toUpperCase());
+}
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply plugins/auth-keycloak/src/utils/helpers.ts]: Added `js-convert-case` package as `change-case` was causing problems with `jest`.

**Comment by Shurtu-gal** [comment]
@overbit The required changes are done.

**Review by overbit — APPROVED** [review]

**Pull request merged**